src/ripple/rpc/handlers/Tx.cpp

```diff
@@ -144,4 +152,107 @@ Json::Value doTx (RPC::Context& context)
     return ret;
 }
 
-} // ripple
+std::pair<rpc::v1::GetTxResponse, grpc::Status>
+doTxGrpc(RPC::ContextGeneric<rpc::v1::GetTxRequest>& context)
+{
+    // return values
+    rpc::v1::GetTxResponse result;
+    grpc::Status status = grpc::Status::OK;
+
+    // input
+    rpc::v1::GetTxRequest& request = context.params;
+
+    std::string const& hashBytes = request.hash();
+    uint256 hash = uint256::fromVoid(hashBytes.data());
+
+    // hash is included in the response
+    result.set_hash(request.hash());
+
+    // get the transaction
+    std::shared_ptr<Transaction> txn =
+        context.app.getMasterTransaction().fetch(hash, true);
+    if (!txn)
+    {
+        grpc::Status errorStatus{grpc::StatusCode::NOT_FOUND, "txn not found"};
+        return {result, errorStatus};
+    }
+
+    std::shared_ptr<STTx const> stTxn = txn->getSTransaction();
+    if (stTxn->getTxnType() != ttPAYMENT)
+    {
+        auto getTypeStr = [&stTxn]() {
+            return TxFormats::getInstance()
+                .findByType(stTxn->getTxnType())
+                ->getName();
+        };
+
+        grpc::Status errorStatus{grpc::StatusCode::UNIMPLEMENTED,
+                                 "txn type not supported: " + getTypeStr()};
+        return {result, errorStatus};
+    }
+
+    // populate transaction data
+    if (request.binary())
+    {
+        Serializer s = stTxn->getSerializer();
+        result.set_transaction_binary(s.data(), s.size());
+    }
+    else
+    {
+        RPC::populateTransaction(*result.mutable_transaction(), stTxn);
+    }
+
+    result.set_ledger_index(txn->getLedger());
+
+    std::shared_ptr<Ledger const> ledger =
+        context.ledgerMaster.getLedgerBySeq(txn->getLedger());
+    // get meta data
+    if (ledger)
+    {
+        if (request.binary())
+        {
+            SHAMapTreeNode::TNType type;
+            auto const item = ledger->txMap().peekItem(txn->getID(), type);
+
+            if (item && type == SHAMapTreeNode::tnTRANSACTION_MD)
+            {
+                SerialIter it(item->slice());
+                it.getVL();  // skip transaction
```

Comment:
`getVL` allocates a buffer we never use. Maybe `it.skip(it.getVLDataLength())` instead?

Reply:
changed to `it.skip(...)`